### PR TITLE
feat: Add `datagram::Batch::extend_from_slice()`

### PR DESCRIPTION
**Pull request by larseggert**
Makes WebRTC reuse of `datagram::Batch` for GSO easier.

### CODE REVIEW

File: neqo-common/src/datagram.rs [hunk 1]
```diff
@@ -256,6 +256,10 @@ impl Batch {
         })
     }
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply neqo-common/src/datagram.rs:258]: In addition to the suggestion from Copilot here, I think that you want to be very careful about the potential for a latent bug.

The use of `slice::chunks_mut` in the iterator creates a situation where you can have the last datagram in a batch be a different size from the others.  Appending bytes this way might lead people to believe that this adds one datagram or another batch.  Or that you could do `batch1.extend_from_slice(batch2.as_slice())` after just checking that the sizes match.  That's not going to be the case, so you probably want to a) document this limitation very clearly OR b) don't allow extension when the last datagram isn't full-sized.

Either way, you want some tests that demonstrate this corner case.

**Review by martinthomson — CHANGES_REQUESTED** [review]

**Comment by larseggert** [comment]
I'll leave this as draft per @mxinden's suggestion to first stabilize the Gecko parts.

File: neqo-common/src/datagram.rs [hunk 2]
```diff
@@ -256,6 +256,20 @@ impl Batch {
...
 
+    /// Append a single datagram's payload to the batch.
+    ///
+    /// Returns `true` if appended successfully, `false` if rejected because:
+    /// - `data` is empty or larger than `datagram_size`, or
+    /// - the last datagram in the batch is partial (not yet full-sized).
+    pub fn push_datagram(&mut self, data: &[u8]) -> bool {
```
Comment (larseggert) [inline neqo-common/src/datagram.rs:264]: Should this return `Result` instead?

File: neqo-common/src/datagram.rs [hunk 3]
```diff
@@ -256,6 +256,20 @@ impl Batch {
         })
     }
 
+    /// Append a single datagram's payload to the batch.
+    ///
+    /// Returns `true` if appended successfully, `false` if rejected because:
+    /// - `data` is empty or larger than `datagram_size`, or
+    /// - the last datagram in the batch is partial (not yet full-sized).
```
Comment (mxinden) [inline neqo-common/src/datagram.rs:263]: How can a datagram _not yet_ be full-sized? Do you expect a user to call `push_datagram` multiple times for a single on-the-wire datagram? If so, is that intuitive?

**Review by mxinden — COMMENTED** [review]
Do I understand correctly, that this is a pure convenience function, in other words that a user could as well aggregate datagrams in its own `Vec<u8>` and only as a final step create a `DatagramBatch`? If so, is this helper function worth the potential foot guns it introduces?

**Comment by larseggert** [comment]
Excellent catch @mxinden. No need for this.
